design_drafts/controller_components.md

```diff
@@ -0,0 +1,48 @@
+# Loading Controller as Components
+
+## Components in a Nutshell
+
+ROS2 Eloquent introduced the concept of components.
+A component represents a node which can be conveniently loaded into a generic components container and so being executed without the node to compile the node in its own executable.
+Throughout the rest of this document, the word `node` is equivalent to `component`.
+The component container (a.k.a `ComponentManager`) takes ownership over the loaded components and can thus control their life cycle, in which components can be dynamically loaded and unloaded.
```

Comment:
Non-native speaker: does this sentence state that components may be (dynamically) loaded and unloaded "in their life cycle", or in the component container?

Reply:
the latter. I'll rewrite the sentence.